src/include/migraphx/op/reshape.hpp

```diff
@@ -46,14 +47,66 @@ struct reshape
     value attributes() const { return {{"require_std_shape", true}}; }
 
     std::string name() const { return "reshape"; }
-    shape compute_shape(std::vector<shape> inputs) const
+
+    shape dyn_compute_shape(shape s0) const
     {
-        check_shapes{inputs, *this}.has(1).standard();
+        auto dyn_dims       = s0.dyn_dims();
+        int not_fixed_index = -1;
+        // track number of fixed elements in input and output
+        std::size_t num_dims_ele = 1;
+        std::size_t num_dd_ele   = 1;
+        for(std::size_t i = 0; i < dyn_dims.size(); ++i)
+        {
+            if(dyn_dims[i].is_fixed())
+            {
+                num_dims_ele *= dims[i];
+                num_dd_ele *= dyn_dims[i].min;
+            }
+            else
+            {
+                if(not_fixed_index == -1)
+                {
+                    not_fixed_index = i;
+                }
+                else
+                {
+                    MIGRAPHX_THROW("Reshape: Only support one non-fixed dynamic_dimension");
+                }
```

Comment:
Use `count_if` to check for multiple non-fixed dynamic dimensions.

Reply:
I changed it, but essentially added a loop over the `dyn_dims`. Doing everything in the `count_if` loop seemed hard to read.